pkg/plugins/config.go

```diff
@@ -2193,6 +2245,69 @@ func (l *Label) mergeFrom(other *Label) error {
 	return utilerrors.NewAggregate(errs)
 }
 
+// Merge two Blunderbuss configurations, returning an aggregated error for conflicts
+func (b *Blunderbuss) mergeFrom(other *Blunderbuss) error {
+	// No config actually specified, so no action required
+	if other == nil {
+		return nil
+	}
+
+	var errs []error
+
+	// Handle global configs
+	if other.BlunderbussConfig != nil {
+		if b.BlunderbussConfig == nil {
+			b.BlunderbussConfig = other.BlunderbussConfig
+		} else if !reflect.DeepEqual(b.BlunderbussConfig, other.BlunderbussConfig) {
+			// Add error when both configs declare different global defaults
+			errs = append(errs, fmt.Errorf("global configurations for blunderbuss do not match"))
+		}
+	}
+
+	// Initialize the Orgs map if it's empty to accept incoming Orgs configs
+	if other.Orgs != nil && b.Orgs == nil {
+		b.Orgs = map[string]BlunderbussOrgConfig{}
+	}
+
+	// Merge Orgs configs, skipping conflicts and adding a message to the aggregated error message
+	for org, otherOrgConfig := range other.Orgs {
+		if orgConfig, ok := b.Orgs[org]; ok {
+			// Use incoming org if current org config is empty, otherwise verify they're the same
+			if otherOrgConfig.BlunderbussConfig != nil {
+				if orgConfig.BlunderbussConfig == nil {
+					orgConfig.BlunderbussConfig = otherOrgConfig.BlunderbussConfig
+					b.Orgs[org] = orgConfig
+				} else if !reflect.DeepEqual(&orgConfig.BlunderbussConfig, &otherOrgConfig.BlunderbussConfig) {
```

Comment:
Are we comparing here values or pointers?

Reply:
Oops--no it doesn't! Let me fix these. Good catch.